samples/timer/src/main.c

```diff
@@ -0,0 +1,108 @@
+/*
+ * Copyright (c) 2022 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: BSD-3-Clause
+ */
+
+#include <zephyr/logging/log.h>
+#include <zephyr/kernel.h>
+#include <npmx_core.h>
+#include <npmx_driver.h>
+#include <npmx_timer.h>
+
+#define LOG_MODULE_NAME timer
+LOG_MODULE_REGISTER(LOG_MODULE_NAME);
+
+/* Timeout in milliseconds during which an npmx device should call a timer interrupt. */
+#define APP_TIMEOUT_MILLISECONDS 500
+
+/* Structure used to submit @ref timeout_callback after the @ref APP_TIMEOUT_MILLISECONDS delay. */
+static struct k_work_delayable timeout_timer;
+
+/**
+ * @brief Callback function called if npmx device does not call timer interrupt.
+ *
+ * @param[in] work Structure used to submit work.
+ */
+static void timeout_callback(struct k_work *work)
+{
+	LOG_ERR("Timeout, interrupt not received within %d ms.", APP_TIMEOUT_MILLISECONDS);
+}
+
+/* Variable filled in @ref main() when timer starts working. */
+static volatile uint32_t start_time;
+
+/**
+ * @brief Callback function to be used when SHIPHOLD WATCHDOG event occurs.
+ *        SHIPHOLD WATCHDOG is used by the timer in the general purpose timer mode.
+ *
+ * @param[in] p_pm Pointer to the instance of nPM device.
+ * @param[in] type Callback type. Should be @ref NPMX_CALLBACK_TYPE_EVENT_SHIPHOLD.
+ * @param[in] mask Received event interrupt mask.
+ */
+static void timer_callback(npmx_instance_t *p_pm, npmx_callback_type_t type, uint8_t mask)
+{
+	/* Cancel timeout. */
+	k_work_cancel_delayable(&timeout_timer);
+
+	/* Calculate elapsed time and print log info. */
+	LOG_INF("Elapsed time: %d ms.", k_uptime_get_32() - start_time);
+
+	/* Call generic callback to print interrupt data. */
+	p_pm->generic_cb(p_pm, type, mask);
```

Comment:
Not necessary here?

Reply:
It prints:
```
[00:00:00.856,781] <dbg> NPMX: generic_callback: SHIPHOLD:
[00:00:00.856,811] <dbg> NPMX: generic_callback:        Watchdog Timeout Warn
```

In my opinion, it is worth having it here.